clients/src/main/java/org/apache/kafka/common/serialization/ListSerializer.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.serialization;
+
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.common.utils.Utils;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class ListSerializer<Inner> implements Serializer<List<Inner>> {
+
+    private Serializer<Inner> inner;
+    private boolean isFixedLength;
+
+    static private List<Class<? extends Serializer<?>>> fixedLengthSerializers = Arrays.asList(
+        ShortSerializer.class,
+        IntegerSerializer.class,
+        FloatSerializer.class,
+        LongSerializer.class,
+        DoubleSerializer.class,
+        UUIDSerializer.class);
+
+    public ListSerializer() {}
+
+    public ListSerializer(Serializer<Inner> serializer) {
+        this.inner = serializer;
+        this.isFixedLength = fixedLengthSerializers.contains(serializer.getClass());
+    }
+
+    @SuppressWarnings(value = "unchecked")
+    @Override
+    public void configure(Map<String, ?> configs, boolean isKey) {
+        if (inner == null) {
+            final String innerSerializerPropertyName = isKey ? CommonClientConfigs.DEFAULT_LIST_KEY_SERIALIZER_INNER_CLASS : CommonClientConfigs.DEFAULT_LIST_VALUE_SERIALIZER_INNER_CLASS;
+            final Object innerSerializerClassOrName = configs.get(innerSerializerPropertyName);
+            try {
+                if (innerSerializerClassOrName instanceof String) {
```

Comment:
Maybe add a `null` check and throw `ConfigException` with detailed error message similar to the `null`-check for `listClass` in the `ListDeserializer#configure(...)`?

I think `instanceof` would be false for `null` and thus the `null` check within `Utils.newInstance(...)` would not be executed.